### DISCUSSION TITLE
Using dynamic database when loaded onto client

### DISCUSSION THREAD

**Discussion by AkshayPathak** [prisma/prisma#5201] [Q&A] [answered]
Hi there,

I would like to know if Prisma 2 supports using dynamic databases. 

For example, I would like to ship an app using Electron and Prisma. The database I would like to use is PostgreSQL. How would the client be able to dynamically point the database to the server running on their machine though a settings menu?

I have seen ideas suggesting environment variables but a change in environment variables would mean that the Prisma client would have to get re-generated. Is my assumption on that correct?

Is there any way I can achieve this with Prisma 2?

Thanks for all the help and effort with creating and maintaining this project.

**Comment by ryands17** [comment, marked as answer]
Hey @AkshayPathak 👋 
In this case, instead of environment variables, you can directly pass the URL in the `PrismaClient` options as follows:

```ts
let prisma = new PrismaClient({
    datasources: { db: { url } },
})
```

So now you just would need to call `prisma.$disconnect()` and assign the `prisma` variable a new `PrismaClient` instance when the database is switched.

If you just want to allow switching between Postgres databases, and the schema is the same then Prisma Client doesn't need to be generated again, otherwise you would need to run `prisma generate` as well.

Let me know if that answers your question :)

**Reply by AkshayPathak** [reply]
@ryands17 Thank you so much, the solution is perfect!

**Comment by escbooster12-dev** [comment]
While the solution given by @ryands17 was working. My question though, is how about if the other database was not yet migrated. How can we perform dynamic database **with/that auto migrates the database?** Thanks and will appreciate any reply :bow:

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
